src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceProperties.cs

```diff
@@ -0,0 +1,199 @@
+using System.Linq;
+using System.Net.Sockets;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPInterfaceProperties : IPInterfaceProperties
+    {
+        private readonly LinuxNetworkInterface _linuxNetworkInterface;
+        private readonly string _dnsSuffix;
+        private readonly UnicastIPAddressInformationCollection _unicastAddresses;
+        private readonly MulticastIPAddressInformationCollection _multicastAddreses;
+        private readonly IPAddressCollection _dnsAddresses;
+        private readonly GatewayIPAddressInformationCollection _gatewayAddresses;
+        private readonly LinuxIPv4InterfaceProperties _ipv4Properties;
+        private readonly LinuxIPv6InterfaceProperties _ipv6Properties;
+
+        public LinuxIPInterfaceProperties(LinuxNetworkInterface lni)
+        {
+            _linuxNetworkInterface = lni;
+            _dnsSuffix = GetDnsSuffix();
+            _unicastAddresses = GetUnicastAddresses();
+            _multicastAddreses = GetMulticastAddresses();
+            _dnsAddresses = GetDnsAddresses();
+            _gatewayAddresses = GetGatewayAddresses();
+            _ipv4Properties = new LinuxIPv4InterfaceProperties(_linuxNetworkInterface);
+            _ipv6Properties = new LinuxIPv6InterfaceProperties(_linuxNetworkInterface);
+        }
+
+        public override bool IsDnsEnabled
+        {
+            get
+            {
+                // /etc/nsswitchc.conf may be of use here.
+                // the absence of "dns" in that file could indicate that DNS is not enabled.
+                return DnsAddresses.Count > 0;
+            }
+        }
+
+        public override string DnsSuffix { get { return _dnsSuffix; } }
+
+        public override bool IsDynamicDnsEnabled
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override UnicastIPAddressInformationCollection UnicastAddresses { get { return _unicastAddresses; } }
+
+        public override MulticastIPAddressInformationCollection MulticastAddresses { get { return _multicastAddreses; } }
+
+        public override IPAddressInformationCollection AnycastAddresses
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override IPAddressCollection DnsAddresses { get { return _dnsAddresses; } }
+
+        public override GatewayIPAddressInformationCollection GatewayAddresses { get { return _gatewayAddresses; } }
+
+        public override IPAddressCollection DhcpServerAddresses
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override IPAddressCollection WinsServersAddresses
+        {
+            get
+            {
+                // TODO: Can we return false? Is this Windows-specific?
+                throw new PlatformNotSupportedException();
```

Comment:
> I believe this property returns the list of WINS addresses you are configured to use, right?

I recommend adding a functional test to validate that.

Also, this applies to other places in the code where WINS was marked as not supported on *NIX, not only this property.